packages/compiler/src/server/compile-service.ts

```diff
@@ -140,6 +143,15 @@ export function createCompileService({
       ...(additionalOptions ?? {}),
     };
 
+    // If emit is set in additionalOptions, use this setting first
+    // otherwise, obtain the `typespec.lsp.emit` configuration from clientConfigsProvider
+    if (additionalOptions?.emit === undefined) {
+      const configEmits = clientConfigsProvider?.get<string[]>("typespec.lsp.emit");
```

Comment:
clientConfigsProvider?.config?.lsp?.emit 

I think would be a simpler experience